### ISSUE TITLE
feat: skip special character parsing

### ISSUE THREAD

**Issue by ph-fritsche** [testing-library/user-event#586]
@ph-fritsche Maybe add an option to skip detection of key descriptors? Something like:
```js
userEvent.type(input, '{"foo":"bar"}', { skipSpecialCharacters: true })
```

_Originally posted by @lynxtaa in https://github.com/testing-library/user-event/issues/584#issuecomment-800922956_

---

An alternative might be to export this workaround as a utility:
```js
userEvent.type(input, '{"foo":"bar"}'.replace(/[{[]/g, '$&$&'))
// would become
userEvent.type(input, escapeKeyboardInput('{"foo":"bar"}'))
```

**Comment by SiebeVE** [comment]
When typing a `:` or a `/`, it is also escaped/removed from the input value.
See: https://codesandbox.io/s/user-event-type-url-sc8by?file=/src/__tests__/App.test.js

Escaping them with the provided regex helps with the `:` but not with the `/`

**Comment by ph-fritsche** [comment]
@SiebeVE Thanks for the report. The behavior described by you is unintended. A fix is applied in #588 .

**Comment by nickmccurdy** [comment]
Should this be closed? I noticed the original issue (#587) is.

**Comment by ph-fritsche** [comment]
> Should this be closed? I noticed the original issue (#587) is.
Nope. I've extracted that comment to a new issue because there was a bug parsing `/`.

This issue is about adding a feature to ease escaping characters for `type` and `keyboard`.
